### ISSUE TITLE
Image doesn't show up on the website. Just keeps Loading forever.

### ISSUE THREAD

**Issue by uzair789** [ucbdrive/scalabel#19]
Hi, 

I created a folder named images in app/src and added a path to that image in a yml file. Below is the first line of my yml file
            
     - {url: ./images/pegs.png }

It just shows loading on the screen. I am working on ubuntu 14.04 with a chrome browser. I have a screenshot attached with the erors. I think the file is not visible to the server. When I was setting it up, I did 

    docker pull scalabel/server

Do I need to run the build command?

![scalabel_debug](https://user-images.githubusercontent.com/8228949/43347005-9902488e-91c1-11e8-90ad-800041dbfba0.png)

**Comment by xanderchf** [comment]
Did you specify a correct attributes file? This error may be caused by failing to load the attributes.

**Comment by JerryJiehanWang** [comment]
I also have this issue. Does anyone has any ideas?

**Comment by JerryJiehanWang** [comment]
I got the 404 not found error. But I'm sure it's in the directory.
<img width="1610" alt="screen shot 2018-07-28 at 3 55 44 pm" src="https://user-images.githubusercontent.com/33852873/43361247-d0db6dea-927e-11e8-8de4-089484345f59.png">

**Comment by uzair789** [comment]
I copied the same attributes file and categories file as in the examples folder. It still doesnt solve the problem. 
My categories.yml file looks like this

    - name: person
    - name: rider

And my attributes.yml file looks like this

    -
      name: Occluded
      toolType: switch
      tagText: o
    -
      name: Truncated
      toolType: switch
      tagText: t
    -
      name: Traffic Light Color
      toolType: list
      values:
        - NA
        - G
        - Y
        - R
      tagPrefix: t
      tagSuffixes:
        - ""
        - g
        - y
        - r
      buttonColors:
        - white
        - green
        - yellow
        - red


I have the screenshot of the new window attached. I can see the categories and the attributes being pulled on the tool but again, the image doesnt load.

![scalabel_attributes_bug](https://user-images.githubusercontent.com/8228949/43614987-a9670b44-9683-11e8-9491-7d4a78dea055.png)

**Comment by seanremy** [comment]
Can you check the console and see if the error has changed? If it has, could you post a screenshot?

Additionally, could you post your items.yml and the location you're putting them in the project directory?

**Comment by uzair789** [comment]
Hi Sean. Thanks for getting back. So I put the image into 

     scalabel/app/src/images

And my imagelist.yml is on the path (don't think the path for the yml matter though)

     scalabel/

The imagelist.yml looks like this

     - {url: ./images/pegs.png }

Ill add the screenshot of the console. From what I understand from the error mssg, it can't see the image in the app/src/images folder.

![scalabel_debug2](https://user-images.githubusercontent.com/8228949/43659401-ed5b368a-9729-11e8-93fb-688cefc5dbab.png)

**Comment by seanremy** [comment]
Aha! That's the problem. The imagelist.yml needs to have URL paths to the resources, not relative paths / filepaths. Should be something like {url: localhost:<PORT_IN_CONFIG>/images/pegs.png }

Let me know if this fixes your error.

**Comment by JerryJiehanWang** [comment]
It seems like I can't access my localhost in my mac. Is there a way to know my localhost port?

**Comment by seanremy** [comment]
Your port is specified in the config file you pass to the server when you run it. The default in our configs is 8686.

**Comment by JerryJiehanWang** [comment]
Yeah I see the server tries to access localhost:8686 datapath, but it gives me an 404 error. I don't know why.

**Comment by uzair789** [comment]
I changed the path to 

    - {url: localhost:8686/images/pegs.png}

It still doesn't work. I see the same error. Its wierd because I went into app/src/img folder and relabellled the loading.png image to loading1.png. When I do
    
     localhost:8686/img/loading.png

I still see the loading image. I even tried adding the pegs.png into the /img folder but it still doesn't work. Somehow, I feel the changes I am making are not visible to the server.

**Comment by AVCarreiro** [comment]
Also having the same problem. 

The items list is referring to the folder app/src/video01 with the localhost but still can't load ("Failed to load resource: net::ERR_UNKNOWN_URL_SCHEME")

First item:

- {url: localhost:8686/video01/video_01-0000001.jpg}

**Comment by ychen1030** [comment]
Please try adding `http://` in front of localhost, should look something like: 
`- {url: 'http://localhost:8686//video01/video_01-0000001.jpg'}`

**Comment by AVCarreiro** [comment]
I had tried that before, returns 404 not found, although I confirmed the folder and file exist. The folder is under app/src/

Tried removing numbers from the folder name, just in case, same result.

**Comment by seanremy** [comment]
In order to diagnose this, could you please provide me 1) your config.yml 2) the directory structure of the items you've added to your app/src directory and 3) the item_list.yml (or at least a snippet from it). I'll try and replicate it and fix it on my end.

**Comment by seanremy** [comment]
Try replacing 'src: "."' with the full path to your source directory.

**Comment by seanremy** [comment]
I think I may see what's going on. Is the location of the executable on a different drive than the source directory?

**Comment by AVCarreiro** [comment]
But now, replaced src with "." again, and not only I can create the project, but it can also load the images correctly... Weird, but OK.

**Comment by seanremy** [comment]
Weird. I suspect it may have something to do with mounting the drive (since your source directory is in /media/...)

**Comment by seanremy** [comment]
Well I'm glad it works! Please update here if the issue shows back up.

**Comment by anmolkabra** [comment]
Hi @seanremy I'm getting a similar issue. I store the images in app/dist/images/ and the image_list.yml in that directory has URLs in this format: `url - url: 'http://localhost:8686/images/a.jpg'`. I am trying to create a bounding box experiment but the images return 404 not found. I'm using the default config yaml.

I have also tried other variants of the URL: `'localhost:8686/images/a.jpg'`, without apostrophes, and full absolute path to the files. 

The server seems to request `/images/a.jpg` and other images in the task but the images aren't present. I don't know what I'm doing wrong. I'm following the tutorial here: https://www.scalabel.ai/doc/get-started.html.

![image](https://user-images.githubusercontent.com/22307419/45979330-cb772300-c01c-11e8-8acf-76260458d9e9.png)

**Comment by seanremy** [comment]
Could you attach your item_list.yml?

**Comment by anmolkabra** [comment]
```yml
- {url: 'http://localhost:8686/images/Sakura_Tree_large.jpg'}
- {url: 'http://localhost:8686/images/Swan_large.jpg'}
- {url: 'http://localhost:8686/images/beach_scene.jpg'}
- {url: 'http://localhost:8686/images/image-1465348_960_720.jpg'}
- {url: 'http://localhost:8686/images/img_lights.jpg'}
- {url: 'http://localhost:8686/images/img_snowtops.jpg'}
- {url: 'http://localhost:8686/images/living_room_pic.jpg'}
```

**Comment by cochoa0x1** [comment]
I had the same problem, a workaround was to just spin up another server to host the images and point the urls to there.

in the folder with my images:
`python -m http.server 8001`

and then in my image_list.yml
`- {url: http://localhost:8001/image.jpg}`

**Comment by aaman-ee** [comment]
For those using the Windows version, for serving in the same domain, put your image dir with the respective relative (in the config file) under the path app/src

**Comment by lonicera-yx** [comment]
My problem is a little different but of the same essence. 

It just says "Image ... was not found"

I've tried several ways about where to put the images and what to write in the url. 

At first, I put the images in disc E, and in my `image_list.yml`
`- {url: 'E:\datasets\images\1.jpg'}`
Failed!

Then, I put the images in `app/src`, and my `image_list.yml`
`- {url: 'http://localhost:8686/images/1.jpg'}` 
Failed!

I even tried `app/dist` in replacement of `app/src`.
Still, it doesn't work!

I suppose there's sth wrong with loading the images into the server. I'll appreciate it if someone finds the solution to it.

**Comment by cooleel** [comment]
Only @cochoa0x1 's solution works.
Thanks!

**Comment by davidmorosini** [comment]
Thanks @cochoa0x1 !! 

@lonicera-yx , my problem was the same as yours, I solved using the solution provided by @cochoa0x1  

On your  path with images, run:
`python3 -m http.server 8001`

Result:
`$ python3 -m http.server 8001`
`Serving HTTP on 0.0.0.0 port 8001 (http://0.0.0.0:8001/) ...`

But, I used **image_list.yml** pointing to: 
`- {url: http://0.0.0.0:8001/1.jpg}`

**Comment by dionysos4** [comment]
For me it worked when I copied the images to app/dist/<image_dir>/<image_name> and set in my **image_list.yml** - {url: 'http://localhost:8686/<image_dir>/<image_name>'}.

FYI: In the config.yml I set **src** to /opt/scalabel and **src** to <image_dir>

**Comment by umangs94** [comment]
I fixed this by binding another volume by adding `--volume <path-to-images>:/opt/scalabel/app/dist/data`.

The `run` command is
```
docker run \
    --name scalabel \
    --volume /home/umang/scalabel/data:/opt/scalabel/data \
    --volume <path-to-images>:/opt/scalabel/app/dist/data \
    --publish 8686:8686 \
    scalabel/www /opt/scalabel/bin/scalabel --config /opt/scalabel/data/config.yml
```

and the images are specified in `image_list.yml` as: `- {url: '/data/1.jpg', videoName: 'b'}` and so on.

**Comment by umangs94** [comment]
This can be fixed by adding `--volume /media/images/<dir with images>:/opt/scalabel/app/dist/data` in the docker command. The images can then be accessed with `url: /data/<image>` in image_list.yml.
